### PR TITLE
[User] Add helper function to \User class to validate accessibility

### PR DESCRIPTION
**Pull request by johnsaigle**
## Brief summary of changes

Adds a function to determine whether a user can access another user.

See #6093

### CODE REVIEW

**Comment by johnsaigle** [comment]
@jesscall This should be ready for review now

File: php/libraries/User.class.inc [hunk 1]
```diff
@@ -557,5 +557,28 @@ class User extends UserPermissions
...
+     *
+     * @return bool
+     */
+    public function isAccessibleBy(\User $user): bool
+    {
+        $centers = $user->getCenterIDs();
+
+        $projMatch   = !empty(
+            array_intersect(
+                $this->getProjectID(),
```
Comment (driusan) [inline php/libraries/User.class.inc:575]: This doesn't look like it's defined?

Comment (johnsaigle) [reply php/libraries/User.class.inc:575]: Yeah you're right. I'll update this.

**Comment by johnsaigle** [comment]
@driusan Should be fixed now

File: php/libraries/User.class.inc [hunk 2]
```diff
@@ -557,5 +557,33 @@ class User extends UserPermissions
...
+     * Implements the AccessibleResource interface. A user can
+     * be accessed by another user if they have a project and a site in common.
+     *
+     * @param \User $user The "editing" user trying to access this user object.
+     *
+     * @return bool
+     */
+    public function isAccessibleBy(\User $user): bool
+    {
+        $centers = $user->getCenterIDs();
```
Comment (jesscall) [inline php/libraries/User.class.inc:571]: this isn't being used anywhere

Comment (johnsaigle) [inline php/libraries/User.class.inc:571]:
```suggestion
```

**Pull request merged**